userspace/libsinsp/sinsp_filter_transformer.h

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2024 The Falco Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include <vector>
+#include <string>
+#include <cstdint>
+#include <functional>
+#include <driver/ppm_events_public.h>
+#include <libsinsp/sinsp_exception.h>
+
+struct extract_value_t
+{
+	uint8_t* ptr = nullptr;
+	uint32_t len = 0;
+};
+
+enum filter_transformer_type: uint8_t
+{
+	FTR_TOUPPER = 0,
+	FTR_TOLOWER = 1,
+	FTR_BASE64 = 2,
+	FTR_STORAGE = 3, // This transformer is only used internally
```

Comment:
for the reviewers `FTR_STORAGE` is just used internally for plugins, more details in the next PRs